### PR TITLE
Fix demos compile failed

### PR DESCRIPTION
**Pull request by imlidian**
```
demo_imi_pq.cpp:138:33: error: cannot initialize a parameter of type 'const faiss::Index::idx_t *' (aka 'const long long *')
      with an rvalue of type 'std::__1::vector<long, std::__1::allocator<long> >::value_type *' (aka 'long *')
                                ids.data() + begin);
```

https://github.com/facebookresearch/faiss/pull/1320

### CODE REVIEW

**Comment by mdouze** [comment]
Maybe hold on, we are in the process of moving the build chain to cmake.

File: demos/demo_imi_pq.cpp [hunk 1]
```diff
@@ -121,7 +121,7 @@ int main ()
                 elapsed() - t0, nb);
 
         std::vector <float> database (nb * d);
-        std::vector <long> ids (nb);
+        std::vector <int64_t> ids (nb);
```
Comment (beauby) [inline demos/demo_imi_pq.cpp:124]: This should technically be `faiss::Index::idx_t`.

Comment (imlidian) [reply demos/demo_imi_pq.cpp]: thins for advice

**Comment by beauby** [comment]
@imlidian  Please rebase and we can merge this.

**Comment by mdouze** [comment]
This was probably fixex.

**Pull request closed without merging**